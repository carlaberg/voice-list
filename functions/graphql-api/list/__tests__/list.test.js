const { lists } = require('../../../fixtures/entities/lists');
const { listItems } = require('../../../fixtures/entities/listitems');
const TestDatabase = require('../../test-utils/TestDatabase')
const db = new TestDatabase()
const List = require('../list.model')
const GQLTestClient = require('../../test-utils/GQLTestCLient')
const gql = require('graphql-tag');
const { fakeList1 } = require('../../../fixtures/entities/lists')

beforeAll(async () => {
  await db.connect();
  await db.seed('lists', lists)
  await db.seed('listitems', listItems)
})

afterAll(async () => {
  await db.stop();
})
 
describe('List resolvers', () => {
  it('Can retrieve single list by list ID', async () => {
    const GET_LIST = `
      query List($id: ID!) {
        list(id: $id) {
          _id
          name
          items {
            text
          }
          createdBy {
            _id
          }
        }
      }
    `

    const { query, mutate } = GQLTestClient()
    res = await query({
      query: GET_LIST,
      variables: { id: fakeList1.stringId }
    })

  });
});