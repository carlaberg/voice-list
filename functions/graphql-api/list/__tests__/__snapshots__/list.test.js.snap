// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List resolvers Can retrieve single list by list ID 1`] = `
Object {
  "data": Object {
    "list": Object {
      "_id": "5f54d28ebef1e6f60116380b",
      "createdBy": Object {
        "_id": "5f54d28ebef1e6f60116380c",
      },
      "items": Array [
        Object {
          "text": "m%o[omsj",
        },
        Object {
          "text": "#uy(4giiycum#hn",
        },
      ],
      "name": "Johanna",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
